pytestqt/wait_signal.py

```diff
@@ -54,11 +57,76 @@ def _quit_loop_by_timeout(self):
             self._loop.quit()
 
     def _cleanup(self):
+        # store timeout message before the data to construct it is lost
+        self._timeout_message = self.get_timeout_error_message()
         if self._timer is not None:
             _silent_disconnect(self._timer.timeout, self._quit_loop_by_timeout)
             self._timer.stop()
             self._timer = None
 
+    def get_timeout_error_message(self):
+        pass
```

Comment:
Since this must be overridden in subclasses, you should do `raise NotImplementedError` here ([docs](https://docs.python.org/3.5/library/exceptions.html#NotImplementedError)) - that should also mean no coverage is required for it, and if coverage.py doesn't do so automatically, we can [tell it to](https://coverage.readthedocs.io/en/default/excluding.html#advanced-exclusion).